### PR TITLE
Implement connOpenInit datagram in the ICS connection spec

### PR DESCRIPTION
**Pull request by majecty**
This PR depends on ~https://github.com/CodeChain-io/foundry/pull/122~, #161 

Implement ICS connection issue: https://github.com/CodeChain-io/foundry/issues/156

### CODE REVIEW

File: core/src/ibc/transaction_handler/datagrams.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    ConnOpenInit {
+        identifier: String,
+        desired_counterparty_connection_identifier: String,
+        counterparty_prefix: String,
+        client_identifier: String,
+        counterparty_client_identifier: String,
+    },
+}
+
+impl Encodable for Datagram {
```
Comment (junha1) [inline core/src/ibc/transaction_handler/datagrams.rs]: Same

File: core/src/ibc/connection_03/types.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    pub state: ConnectionState,
+    pub counterparty_connection_identifier: Identifier,
+    pub counterparty_prefix: CommitmentPrefix,
+    pub client_identifier: Identifier,
+    pub counterparty_client_identifier: Identifier,
+    // FIXME: implement version
+}
+
+impl rlp::Encodable for ConnectionEnd {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
```
Comment (junha1) [inline core/src/ibc/connection_03/types.rs]: Did you check that RLP **macro** can handle this?

Comment (majecty) [reply core/src/ibc/connection_03/types.rs]: Sure, I'll check them.

Comment (majecty) [reply core/src/ibc/connection_03/types.rs]: I checked this code: https://github.com/CodeChain-io/rlp/blob/64fdbc5758e05483f62fae99521520f566eaca9d/rlp-derive/src/en.rs#L21
It seems it is suitable to use it. I'll try it.

Comment (majecty) [reply core/src/ibc/connection_03/types.rs]: I changed it. Could you check it again?

File: core/src/ibc/transaction_handler/datagrams.rs (hunk 1 above)
Comment (majecty) [reply core/src/ibc/transaction_handler/datagrams.rs]: I changed it. Could you check it again?

**Pull request merged**